pandas/core/frame.py

```diff
@@ -440,6 +442,8 @@ def __init__(self, data=None, index=None, columns=None, dtype=None, copy=False):
             if not isinstance(data, abc.Sequence):
                 data = list(data)
             if len(data) > 0:
+                if is_dataclass_instance(data[0]):
+                    data = _dataclasses_to_dicts(data)
```

Comment:
should this not be *after* the is_named_tuples check (below)?

Reply:
That's nested under an `is_list_like(data[0])`. I don't think dataclasses are considered list like.

Reply:
correct, a `dataclass` is not "list like" so moving that there would just break this PR.